pkg/util/restoptions/helpers.go

```diff
@@ -51,10 +52,14 @@ func ApplyOptions(optsGetter Getter, store *registry.Store, oldIsNamespaced bool
 	if store.CreateStrategy == nil {
 		return fmt.Errorf("store for %s must have CreateStrategy set", store.QualifiedResource.String())
 	}
-
-	isNamespaced := store.CreateStrategy.NamespaceScoped()
-	if isNamespaced != oldIsNamespaced { // TODO(soltysh): oldIsNamespaced should be completely removed in #12541
-		return fmt.Errorf("CreateStrategy has %v for namespace scope but user specified %v as namespace scope", isNamespaced, oldIsNamespaced)
+	if store.ObjectNameFunc == nil {
```

Comment:
`store.CompleteWithOptions` will need to be updated with this upstream.  All of the object name funcs will also need to be deleted.  To do so safely means we need the etcd storage test upstream as well.